src/overlays/actors/ovl_En_Ma_Yto/z_en_ma_yto.c

```diff
@@ -21,126 +83,1425 @@ const ActorInit En_Ma_Yto_InitVars = {
+        case MA_YTO_TYPE_DEFAULT:
+            return 0;
+
+        case MA_YTO_TYPE_DINNER:
+            if (!(gSaveContext.weekEventReg[0x16] & 1) && CURRENT_DAY == 2) {
+                return 0;
+            }
+            if (EnMaYto_SearchRomani(this, globalCtx)) {
+                return 2;
+            }
+            return 1;
+
+        case MA_YTO_TYPE_BARN:
+            // if (AliensDefeated)
+            if (gSaveContext.weekEventReg[0x16] & 1) {
+                if (EnMaYto_SearchRomani(this, globalCtx)) {
+                    return 2;
+                }
+                return 1;
+            }
+            return 0;
+
+        case MA_YTO_TYPE_AFTERMILKRUN:
+            return 0;
+
+        case MA_YTO_TYPE_4:
+            return 0;
+    }
+
+    return 0;
+}
+
+void EnMaYto_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnMaYto* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnMaYto_SetupKeepLookingForRomani(EnMaYto* this) {
+    this->actionFunc = EnMaYto_KeepLookingForRomani;
+}
+
+void EnMaYto_KeepLookingForRomani(EnMaYto* this, GlobalContext* globalCtx) {
+    if (EnMaYto_TryFindRomani(this, globalCtx) == 2) {
+        EnMaYto_ChooseAction(this, globalCtx);
+    }
+}
+
+void EnMaYto_SetupDefaultWait(EnMaYto* this) {
+    if (this->actor.shape.rot.y == this->actor.home.rot.y) {
+        this->currentAnim = 11;
+        EnMaYto_ChangeAnim(this, 11);
+    } else {
+        this->currentAnim = 1;
+        EnMaYto_ChangeAnim(this, 1);
+    }
+
+    EnMaYto_InitFaceExpression(this);
+    this->unk_31E = 2;
+    this->actionFunc = EnMaYto_DefaultWait;
+}
+
+void EnMaYto_DefaultWait(EnMaYto* this, GlobalContext* globalCtx) {
+    s16 rotY = this->actor.home.rot.y - 0x8000;
+    s16 sp2C;
+
+    sp2C = rotY - this->actor.yawTowardsPlayer;
+    if (Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.home.rot.y, 5, 0x3000, 0x100) == 0 &&
+        this->currentAnim == 1) {
+        this->currentAnim = 11;
+        EnMaYto_ChangeAnim(this, 11);
+    }
+
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        EnMaYto_DefaultStartDialogue(this, globalCtx);
+        EnMaYto_SetupDefaultDialogueHandler(this);
+    } else if (ABS_ALT(sp2C) < 0x1555) {
+        func_800B8614(&this->actor, globalCtx, 100.0f);
+    }
+}
+
+void EnMaYto_SetupDefaultDialogueHandler(EnMaYto* this) {
+    EnMaYto_ChangeAnim(this, 1);
+    this->unk_31E = 2;
+    this->actionFunc = EnMaYto_DefaultDialogueHandler;
+}
+
+void EnMaYto_DefaultDialogueHandler(EnMaYto* this, GlobalContext* globalCtx) {
+    switch (func_80152498(&globalCtx->msgCtx)) {
+        case 4:
+            EnMaYto_DefaultHandlePlayerChoice(this, globalCtx);
+            break;
+
+        case 5:
+            EnMaYto_DefaultChooseNextDialogue(this, globalCtx);
+            break;
+
+        case 6:
+            if (func_80147624(globalCtx) != 0) {
+                this->unk_31E = 0;
+                EnMaYto_SetupDefaultWait(this);
+            }
```

Comment:
```suggestion
            if (func_80147624(globalCtx)) {
                this->unk_31E = 0;
                EnMaYto_SetupDefaultWait(this);
            }
```
It's might be better to wait for the decomp of this function before removing the `!= 0`, but all other uses of this function are in if-statements and nearly all of them already removed the `!= 0`. Applies to the other uses of this function throughout the file if you want to change it